iModelCore/ecobjects/test/NonPublished/SchemaMergerTests.cpp

```diff
@@ -5504,55 +5519,442 @@ TEST_F(SchemaMergerTests, TestBaseClassAdditionAndRemoval)
   EXPECT_TRUE(issues.m_issues.empty());
   }
 
-/*---------------------------------------------------------------------------------**//**
+/*---------------------------------------------------------------------------------------
 * @bsitest
 +---------------+---------------+---------------+---------------+---------------+------*/
-/*TEST_F(SchemaMergerTests, MergeSchemasFromLocalDirectories)
+TEST_F(SchemaMergerTests, DuplicateSchemaNamesMergeResult)
     {
-    BeFileName leftFolder(L"f:\\defects\\schemaDump_constraint\\Left\\");
-    BeFileName rightFolder(L"f:\\defects\\schemaDump_constraint\\Right\\");
+    // Initialize two sets of schemas
+    bvector<Utf8CP> leftSchemasXml {
+      R"schema(<?xml version='1.0' encoding='utf-8' ?>
+        <ECSchema schemaName="MYSCHEMA" alias="mys" version="01.00.00" xmlns="http://www.bentley.com/schemas/Bentley.ECXML.3.2">
+          <ECEntityClass typeName="A">
+          </ECEntityClass>
+        </ECSchema>
+        )schema"
+    };
+    ECSchemaReadContextPtr leftContext = InitializeReadContextWithAllSchemas(leftSchemasXml);
+    bvector<ECN::ECSchemaCP> leftSchemas = leftContext->GetCache().GetSchemas();
+
+    bvector<Utf8CP> rightSchemasXml {
+      R"schema(<?xml version='1.0' encoding='utf-8' ?>
+        <ECSchema schemaName="MySchema" alias="mys" version="01.00.01" xmlns="http://www.bentley.com/schemas/Bentley.ECXML.3.2">
+          <ECEntityClass typeName="A">
+          </ECEntityClass>
+          <ECEntityClass typeName="B">
+            <BaseClass>A</BaseClass>
+          </ECEntityClass>
+        </ECSchema>
+        )schema"
+    };
+    ECSchemaReadContextPtr rightContext = InitializeReadContextWithAllSchemas(rightSchemasXml);
+    bvector<ECN::ECSchemaCP> rightSchemas = rightContext->GetCache().GetSchemas();
+
+    // Merge the schemas
+    SchemaMergeResult result;
+    MergerTestIssueListener issues;
+    result.AddIssueListener(issues);
+    EXPECT_EQ(BentleyStatus::ERROR, SchemaMerger::MergeSchemas(result, leftSchemas, rightSchemas));

```

Comment:
I believe you found a new bug here :-) Schema Item and Schema Names are supposed to ignore case. So the result should never treat this input differently as if they had the identical names.... Let's fix this while we're at it. The intended result for this test is Success with the right schema xml as the output.

Reply:
The ECSchema name comparison utility function `CompareByName` explicitly notes case-sensitive checking to be the expected behavior. However. as per what we've discussed, SchemaMerger will now disregard case when merging schemas.